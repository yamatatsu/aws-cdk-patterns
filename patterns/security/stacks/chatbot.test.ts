import { SynthUtils } from "@aws-cdk/assert"
import * as cdk from "@aws-cdk/core"
import * as sns from "@aws-cdk/aws-sns"
import { Chatbot } from "./chatbot"

test("snapshot test", () => {
  const app = new cdk.App()
  const stack = new cdk.Stack(app, "testStack")

  const topic = new sns.Topic(stack, "test-Topic")

  const target = new Chatbot(app, "Target", {
    slackWorkspaceId: "test-slackWorkspaceId",
    slackChannelId: "test-slackChannelId",
    notificationTopics: [topic],
  })

  expect(SynthUtils.toCloudFormation(target)).toMatchSnapshot()
})
