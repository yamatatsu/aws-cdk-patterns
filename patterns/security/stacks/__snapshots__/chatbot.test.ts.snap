// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
Object {
  "Resources": Object {
    "SlackChannelConfigurationC73E0866": Object {
      "Properties": Object {
        "ConfigurationName": "SecurityAlert",
        "IamRoleArn": Object {
          "Fn::GetAtt": Array [
            "SlackChannelConfigurationConfigurationRole066766EC",
            "Arn",
          ],
        },
        "SlackChannelId": "test-slackChannelId",
        "SlackWorkspaceId": "test-slackWorkspaceId",
        "SnsTopicArns": Array [
          Object {
            "Fn::ImportValue": "testStack:ExportsOutputReftestTopic13C1EFB8590E3711",
          },
        ],
      },
      "Type": "AWS::Chatbot::SlackChannelConfiguration",
    },
    "SlackChannelConfigurationConfigurationRole066766EC": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "chatbot.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
